### PR TITLE
removed all deprecated 1.1 code

### PR DESCRIPTION
**Pull request by nabobalis**
Should I add a changelog? I feel like I should.

### CODE REVIEW

**Comment by dstansby** [comment]
Yeah, I would add a breaking changelog detailing what is removed and what the alternatives (if any) are in each case.

**Comment by nabobalis** [comment]
Added one. Probably needs more work.

**Comment by nabobalis** [comment]
Online fails
```
ERROR ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/helio/tests/test_helio.py::test_get_table_names - requests.exceptions.ConnectionError: HTTPConnectionPool(host...
ERROR ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/helio/tests/test_helio.py::test_select_table - requests.exceptions.ConnectionError: HTTPConnectionPool(host='w...
ERROR ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/helio/tests/test_helio.py::test_time_query 
```

File: changelog/4350.breaking.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Removed deprecated functions:
+
+`sunpy.coordinates.frames.Helioprojective.calculate_distance`, alternative is `sunpy.coordinates.frames.Helioprojective.calculate_distance.make_3d`.
+`sunpy.image.coalignment.repair_image_nonfinite`, no alternative.
```
Comment (dstansby) [inline changelog/4350.breaking.rst:4]:
```suggestion
``sunpy.image.coalignment.repair_image_nonfinite`` - if you wish to repair the image, this has to be done manually before calling the various `sunpy.image.coalignment` functions. The ``repair_nonfinite`` keyword argument to ``calculate_shift`` and  ``calculate_match_template_shift`` has been removed.
```

File: changelog/4350.breaking.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+`sunpy.instr.lyra.download_lytaf_database`, no alternative.
```
Comment (dstansby) [inline changelog/4350.breaking.rst:5]:
```suggestion
`sunpy.instr.lyra.download_lytaf_database` - this just downloaded the file at ``http://proba2.oma.be/lyra/data/lytaf/annotation_ppt.db``, which can be done manually.
```

Comment (nabobalis) [reply changelog/4350.breaking.rst]: Fixed.

File: changelog/4350.breaking.rst (hunk 1 above)
Comment (nabobalis) [reply changelog/4350.breaking.rst]: Fixed.

File: changelog/4350.breaking.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Removed deprecated functions:
```
Comment (Cadair) [inline changelog/4350.breaking.rst:1]: As stated in the changelog readme, the removal of deprecated features are "removals" and not "breaking". My logic for this (or rationalisation) is that I think it's important to keep the list of breaking changes down to "things you *need* to deal with".

Comment (nabobalis) [reply changelog/4350.breaking.rst:1]: Changed in a PR.

**Pull request merged**